### PR TITLE
Added Interactive tutorial to the API Store

### PR DESCRIPTION
**Pull request by suthagar23**
## Interactive Tutorial for API Store

This PR contains the code for the Interactive tutorial for API Store. It will help the users who visit the API Store and will guide them to subscribe and invoke the APIs at the first time. This tutorial contains, 

1. How to create an application and how to subscribe that application to an API.
2. How to generate the production Keys.
3. How to invoke an API using access tokens.

image

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-info/template.jag [hunk 1]
```diff
@@ -378,6 +378,59 @@ $(window).load(function(){
...
+                runEnjoyHintScript(enjoyhint_api_info_instance, api_store_subscribe_to_default_application);
+            }
+            else {
+                localStorage.setItem("isNewApplicationClicked",true);
+                runEnjoyHintScript(enjoyhint_api_info_instance, api_store_subscribe_to_new_application);
+            }
+            localStorage.setItem("isApplicationSelectedForSubscribe", true);
+          }
+          else {
+              if(isNewApplicationClicked && isUserCreatedNewApplication) {
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-info/template.jag]: Lets format this code

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-listing-all/template.jag [hunk 2]
```diff
@@ -403,7 +461,23 @@ $(window).load(function(){
         });
     });
 
+function catchTheFirstAPINameForTutorial() {
+    var isNewUserInteractiveTutorialStarted = localStorage.getItem("isNewUserInteractiveTutorialStarted");
+    if(isNewUserInteractiveTutorialStarted) {
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-listing-all/template.jag]: Let's format

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-info/template.jag [hunk 3]
```diff
@@ -378,6 +378,59 @@ $(window).load(function(){
...
+  var isAccessKeyGeneratedForTheUser = localStorage.getItem("isAccessKeyGeneratedForTheUser");
+  var isAPIConsoleCompleted = localStorage.getItem("isAPIConsoleCompleted");
+  var isNewApplicationClicked = localStorage.getItem("isNewApplicationClicked");
+  var isUserCreatedNewApplication = localStorage.getItem("isUserCreatedNewApplication");
+  var enjoyhint_api_info_instance = null;
+
+  if (<%=outputs.user%> && !isTutorialCompleted && isNewUserInteractiveTutorialStarted) {
+      if (!isAccessKeyGeneratedForTheUser && !isApplicationSubscribed ) {
+          if (!isApplicationSelectedForSubscribe && !isNewApplicationClicked) {
+            if (<%=subscribedToDefault%>){
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-info/template.jag]: Formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-listing-all/template.jag [hunk 4]
```diff
@@ -43,14 +45,70 @@ if(length==0){
 }
 
 var reqUrl = request.getRequestURI();
+if (jagg.getUser() != null) {
+	isSuperTenant = jagg.getUser().isSuperTenant;
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-listing-all/template.jag]: code is not formatted properly

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-listing-all/template.jag [hunk 5]
```diff
@@ -43,14 +45,70 @@ if(length==0){
...
+            // auto launch the tutorial for the new users
+            if (<%=apps.length <=1 %>) {
+                launchIntractiveTutorial();
+            }
+        }
+    }
+  }
+ 
+  function launchIntractiveTutorial() { 
+        localStorage.setItem("isNewUserInteractiveTutorialStarted", true);
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/api-listing-all/template.jag]: is the formatting correct here? please check.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/application/application-view/template.jag [hunk 6]
```diff
@@ -132,7 +132,33 @@ for(var i=0;i < outputs.tiers.length; i++){
...
             }
         }
+        else {
+          runIntractiveTutorialForUser();
+        }
     });
+
+    // API Store Intractive Tutorial 
+    function runIntractiveTutorialForUser() { 
+      var enjoyhint_application_view_instance = null;
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/application/application-view/template.jag]: please check naming convention for variables.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/libs/enjoyhint-3.1.0/js/cloud-enjoyhint-functions.js [hunk 7]
```diff
@@ -55,6 +55,59 @@ function removeLocalStorageVariables() {
     }
 }
 
+
+function removeLocalStorageVariablesForNewUser() {
+    var isTutorialCompleted = localStorage.getItem("isTutorialCompleted");
+    if (null != isTutorialCompleted) {
```
[the remark this answers is not in the document]
Comment (suthagar23) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/libs/enjoyhint-3.1.0/js/cloud-enjoyhint-functions.js]: Thanks for the info. I have removed that check and getItem().

**Pull request merged**